runtime/runtime/src/actions.rs

```diff
@@ -921,4 +1122,584 @@ mod tests {
+            &mut state_update,
+            &apply_state,
+            &signed_delegate_action.delegate_action,
+            &mut result,
+        )
+        .expect("Expect ok");
+        assert_eq!(
+            result.result,
+            Err(ActionErrorKind::DelegateActionAccessKeyError(
+                InvalidAccessKeyError::AccessKeyNotFound {
+                    account_id: sender_id.clone(),
+                    public_key: sender_pub_key.clone(),
+                },
+            )
+            .into())
+        );
+    }
+
+    #[test]
+    fn test_delegate_action_key_incorrect_nonce() {
+        let mut result = ActionResult::default();
+        let (_, signed_delegate_action) = create_delegate_action_receipt();
+        let sender_id = signed_delegate_action.delegate_action.sender_id.clone();
+        let sender_pub_key = signed_delegate_action.delegate_action.public_key.clone();
+        let access_key = AccessKey {
+            nonce: signed_delegate_action.delegate_action.nonce,
+            permission: AccessKeyPermission::FullAccess,
+        };
+
+        let apply_state =
+            create_apply_state(signed_delegate_action.delegate_action.max_block_height);
+        let mut state_update = setup_account(&sender_id, &sender_pub_key, &access_key);
+
+        validate_delegate_action_key(
+            &mut state_update,
+            &apply_state,
+            &signed_delegate_action.delegate_action,
+            &mut result,
+        )
+        .expect("Expect ok");
+        assert_eq!(
+            result.result,
+            Err(ActionErrorKind::DelegateActionInvalidNonce {
+                delegate_nonce: signed_delegate_action.delegate_action.nonce,
+                ak_nonce: signed_delegate_action.delegate_action.nonce,
+            }
+            .into())
+        );
+    }
+
+    #[test]
+    fn test_delegate_action_key_nonce_too_large() {
+        let mut result = ActionResult::default();
+        let (_, signed_delegate_action) = create_delegate_action_receipt();
+        let sender_id = signed_delegate_action.delegate_action.sender_id.clone();
+        let sender_pub_key = signed_delegate_action.delegate_action.public_key.clone();
+        let access_key = AccessKey { nonce: 19000000, permission: AccessKeyPermission::FullAccess };
+
+        let apply_state = create_apply_state(1);
+        let mut state_update = setup_account(&sender_id, &sender_pub_key, &access_key);
+
+        validate_delegate_action_key(
+            &mut state_update,
+            &apply_state,
+            &signed_delegate_action.delegate_action,
+            &mut result,
+        )
+        .expect("Expect ok");
+        assert_eq!(
+            result.result,
+            Err(ActionErrorKind::DelegateActionNonceTooLarge {
+                delegate_nonce: signed_delegate_action.delegate_action.nonce,
+                upper_bound: 1000000,
+            }
+            .into())
+        );
+    }
+
+    fn test_delegate_action_key_permissions(
```

Comment:
This misses a test tag

```suggestion
    #[test]
    fn test_delegate_action_key_permissions(
```

Reply:
This is not a test. This is a helper function. It's called from other tests

Reply:
oh of course, sorry never mind then